pkgs/by-name/u-/u-config-unwrapped/package.nix

```diff
@@ -0,0 +1,121 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  noLibc ? stdenv.cc.isGNU && stdenv.hostPlatform.isx86 && stdenv.hostPlatform.isLinux,
+  pkgsBuildBuild,
+  installShellFiles,
+  nix-update-script,
+}:
+
+let
+  inherit (stdenv) hostPlatform;
+  inherit (lib) optionals;
+
+  platform =
+    if noLibc then
+      (
+        with hostPlatform;
+        if isLinux && isx86_32 then
+          "linux_i686"
+        else if isLinux && isx86_64 then
+          "linux_amd64"
+        else
+          throw "unsupported system ${system} for libc‐free version"
+      )
+    else
+      "generic";
+
+  cflags =
+    [
+      "-DPKG_CONFIG_PREFIX=\"\""
+      "-DPKG_CONFIG_LIBDIR=\"\""
+      "-DPKG_CONFIG_SYSTEM_INCLUDE_PATH=\"\""
+      "-DPKG_CONFIG_SYSTEM_LIBRARY_PATH=\"\""
+    ]
+    ++ optionals noLibc (
+      [
+        "-fno-builtin"
+        "-fno-asynchronous-unwind-tables"
+        "-static"
+        "-nostdlib"
+        "-Wl,--gc-sections"
+      ]
+      ++ optionals (stdenv.cc.isGNU) [
+        # avoid generating calls to memset() and memcpy()
+        "-fno-tree-loop-distribute-patterns"
+      ]
+    );
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "u-config";
+  version = "0.33.3";
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  src = fetchurl {
+    url = "https://github.com/skeeto/u-config/releases/download/v0.33.3/u-config-${finalAttrs.version}.tar.gz";
+    hash = "sha256-wtSGgU99/UJ06Ww5+tmorhBx70It+cvfp2EO8fCH9ew=";
+  };
+
+  pkgM4 = fetchurl {
+    url = "https://gitlab.freedesktop.org/pkg-config/pkg-config/-/raw/eb866ade77d933c3237a5c9e2114ca65c3baa537/pkg.m4.in";
```

Comment:
Better to use an updated one from `pkgconf` which has some additional stuff. I've been using [this one](https://github.com/pkgconf/pkgconf/raw/refs/tags/pkgconf-1.9.5/pkg.m4).

But regardless of which one you pick, both of these are licensed under GPL-2+ (with autoconf exception). So if you add it here, you need to add that to the pkg's license as well.